docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_java_dependencies.adoc

```diff
@@ -1,30 +1,106 @@
 
-* The system attempts to build all the dependencies and stores them in container images in an image registry, and `quay.io` is the default.
+. The JVM Build Service analyzes your application and determines the dependencies that the system needs to rebuild.
 
-* When the system has completed all its builds, you can rebuild your application.
+. The JVM Build Service generates `ArtifactBuild` objects to represent each artifact in your application before determining to which repository the artifacts belong. The JVM Build Service tags from which systems the artifacts derive.
 
-* If all the builds are successful, your application's SBOM displays that none of the dependencies came from third-party repositories. However, you may need to troubleshoot when some dependencies fail to build.
+. The system analyzes the repository and uses the results to find a build strategy. For example, if the system cannot find the JDK version to use, the system uses all versions to determine the most effective version to use.
 
-.*Procedure*
+. The system attempts to build all the dependencies and stores them in container images in an image registry, and `quay.io` is the default.
+
+. When the system has completed all its builds, you can rebuild your application.
+
+. If all the builds are successful, your application's SBOM displays that none of the dependencies came from third-party repositories. However, you may need to troubleshoot when some dependencies fail to build.
+
+== Setup
+
+=== Logging into the Server
+
+All interactions with the JVM Build Service require you to be logged into the OpenShift namespace used for your builds. This can be done by copying the login command from https://registration-service-toolchain-host-operator.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/ .
+
+=== Setting up the CLI
+
+Many of the commands below can be setup using the JVM Build Service CLI.
+
+The CLI is currently distributed as a docker image. To use the docker image, create the following alias:
+
+```
+alias jbs='docker run --mount type=bind,source=$HOME/.kube/config,target=/kube --mount type=bind,source=$HOME/.github,target=/root/.github --env KUBECONFIG=/kube  -it --rm   quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+To update the latest version, create an update alias. You can run this alias to pull the latest version of the docker image:
+
+```
+alias update-jbs='docker  pull quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+The CLI supports tab completion and all the commands have a --help option to show usage.
+
+Note that this CLI also requires you to be logged in as described above.
+
+To use commands that modify build recipes (i.e. commands that are used to fix failing builds) you will need to create a `$HOME/.github` file as specified at  https://github-api.kohsuke.org/[].
+
+== Configuring The JVM Rebuild Service
+
+There are three ways to setup the JVM Build Service, they are listed below.
+
+=== Setting up using the CLI
```

Comment:
Minor formatting changed based off Red Hat Modularization standards:

=== Setting up using the CLI

If you do not require explicit quay.io configuration, you can simply setup the JVM Build Service. The JVM Build service automatically creates rebuilds for you and creates a quary repository to store your rebuilt artifacts. 

.Prerequisite 

- You have set up the CLI. 

.Procedure 

-  In the CLI, run the `setup rebuilds` command: < your provided command >.